bin/docker/boot_dev

```diff
@@ -89,7 +89,14 @@ done
 # 9292 unicorn
 # 9405 prometheus exporter
 
-docker pull discourse/discourse_dev:release
+if [ "$(uname -m)" == "aarch64" ]; then
+    ENV_ARGS+=" -e SE_MANAGER_PATH=/usr/local/bin/selenium-manager "
+else
+    # NOTE: we currently (2024-01-17) don’t build an aarch64 image for the dev
+    # env, so this command will pull the existing one targeted at x86_64.
+    docker pull discourse/discourse_dev:release
+fi
```

Comment:
This change is a bit surprising... the comment talks about pulling the image or `aarch64`... but the logic is in the `else` block so it won't actually be run for `aarch64`.

I guess it's supposed to be

```suggestion
if [ "$(uname -m)" == "aarch64" ]; then
    ENV_ARGS+=" -e SE_MANAGER_PATH=/usr/local/bin/selenium-manager "
fi

# NOTE: we currently (2024-01-17) don’t build an aarch64 image for the dev
# env, so this command will pull the existing one targeted at x86_64.
docker pull discourse/discourse_dev:release
```

Reply:
Ah, sorry if the comment is confusing :confused: 
We don’t want to call `docker pull` on `aarch64` because it will pull the only available image we have for `discourse/discourse_dev:release` which is the `x86_64` one. It will then overwrite the locally built one you need to have, and it won’t work.

Reply:
Ahhh I see. Can I suggest some slight tweaks to the comments then?

```suggestion
if [ "$(uname -m)" == "aarch64" ]; then
    # NOTE: we currently (2024-01-17) don’t build an aarch64 image, so one must be pre-built locally.
    # Avoiding `docker pull` so we don't override that local image.
    ENV_ARGS+=" -e SE_MANAGER_PATH=/usr/local/bin/selenium-manager "
else
    # x86_64 environment - pull the latest image from dockerhub
    docker pull discourse/discourse_dev:release
fi
```

BTW, on ARM macOS machines, pulling and using the `x86_64` image works fine (albeit a bit slower because of the emulation). Does it fail completely on `aarch64`?

Reply:
That’s better indeed, thanks! :blush: 

Regarding the `x86_64` image, by default, yes it fails completely. The image will be downloaded, but then when you try to run it, it will fail, saying something like `bad exec format`.
Now, I don’t remember if I tried with the qemu/binfmt support or not :thinking: This should provide x86 emulation, I don’t know the cost though :sweat_smile: